pkgs/by-name/ru/rutabaga-gfx-ffi/package.nix

```diff
@@ -0,0 +1,49 @@
+{ aemu
+, rustPlatform
+, fetchFromGitHub
+, pkg-config
+, gfxstream
+, libdrm
+, lib
+}:
+
+rustPlatform.buildRustPackage {
+  pname = "rutabaga_gfx_ffi";
+  version = "unstable-2023-12-20";
+
+  src = fetchFromGitHub {
+    owner = "google";
+    repo = "crosvm";
+    rev = "46279a0c03f8892f22ebd8c0dc19e4a6dc8aac41";
+    hash = "sha256-xHT3uWPGVqXFr5G08jliti5RXSsZLb6fH8eiy8cyzNY=";
+    fetchSubmodules = true;
+  };
+
+  patches = [
+    ./0001-rutabaga_gfx-don-t-clone-wayland-memfd-file-descript.patch
+    ./0002-rutabaga_gfx-super-ugly-workaround-to-get-private-ke.patch
+    ./0003-rutabaga_gfx-fix-stale-cross-domain-keymap-resources.patch
+  ];
+
+  buildPhase = ''
+    cd rutabaga_gfx/ffi
```

Comment:
sourceRoot?

Reply:
We need to point buildRustPackage to the crosvm Cargo.lock or else it won't have all cargo crates fetched. So sourceRoot doesn't work here.

Reply:
We don't need to use buildRustPackage — we can use fetchCargoTarball and let the build system do its thing.

Reply:
Ok. I would say this sounds like it would be about the same complexity of my current solution. I think buildRustPackage is more commonly tested so I will run into less issues.

Reply:
But then we're bypassing upstream's build system.  They're expecting us to build from the Makefile — what if there are important flags etc. in there?  Plus, we wouldn't need the custom phases, so all round less complex.